### PR TITLE
W-17040933-Add-AsyncAPI-notes-to-ACB-docs-CP

### PR DESCRIPTION
**Pull request by Crispy-Salesforce**
W-17040933-Add-AsyncAPI-notes-to-ACB-docs-CP

### CODE REVIEW

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 1]
```diff
@@ -30,20 +30,23 @@ When scaffolding an AsyncAPI specification into a Mule project (implementation o
...
-** *Message Listener* (`<apikit-asyncapi:message-listener/>`): AsyncAPI `publish` operations in the specification are configurable *Message Listener* (_not_ *Publish*) operations. See examples in <<ex-flows-scaffolded>>.
+** *Publish* (`<apikit-asyncapi:publish/>`): AsyncAPI *subscribe* operations in the specification are available for configuration as AsyncAPI *Publish* operations. See the <<ex-flows-implemented, implemented example>> in Example: Scaffolded Flows.
+** *Message Listener* (`<apikit-asyncapi:message-listener/>`): AsyncAPI *publish* operations in the specification are configurable *Message Listener* (_not_ *Publish*) operations. See examples in <<ex-flows-scaffolded>>.
 +
-The scaffolder transforms each `publish` operation into a *Message Listener* operation.
+The scaffolder transforms each *publish* operation into a *Message Listener* operation.
++
+If a *Message Listener* receives an invalid message, it’s sent to the dead-letter queue (you must set up one dead-letter queue).
```
Comment (simoneegeib) [inline modules/ROOT/pages/imp-asyncapi.adoc]: Is this really correct?

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 2]
```diff
@@ -30,20 +30,23 @@ When scaffolding an AsyncAPI specification into a Mule project (implementation o
...
 * Makes the APIkit for AsyncAPI Module (AsyncAPI Module) available for use in the project 
 +
 The module provides the following operations:
 +
-** *Publish* (`<apikit-asyncapi:publish/>`): AsyncAPI `subscribe` operations in the specification are available for configuration as AsyncAPI *Publish* operations. See the <<ex-flows-implemented, implemented example>> in Example: Scaffolded Flows.
-** *Message Listener* (`<apikit-asyncapi:message-listener/>`): AsyncAPI `publish` operations in the specification are configurable *Message Listener* (_not_ *Publish*) operations. See examples in <<ex-flows-scaffolded>>.
+** *Publish* (`<apikit-asyncapi:publish/>`): AsyncAPI *subscribe* operations in the specification are available for configuration as AsyncAPI *Publish* operations. See the <<ex-flows-implemented, implemented example>> in Example: Scaffolded Flows.
+** *Message Listener* (`<apikit-asyncapi:message-listener/>`): AsyncAPI *publish* operations in the specification are configurable *Message Listener* (_not_ *Publish*) operations. See examples in <<ex-flows-scaffolded>>.
 +
-The scaffolder transforms each `publish` operation into a *Message Listener* operation.
```
Comment (simoneegeib) [inline modules/ROOT/pages/imp-asyncapi.adoc:46]: I think we should add the same sentence for 'subscribe' operations. Also, can we link to the AsyncAPI spec to explain why publish is subscribe and vice versa?

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 3]
```diff
@@ -72,7 +75,7 @@ You can scaffold an AsyncAPI specification into a new or existing Mule project:
 [[important-rescaffold]]
 [IMPORTANT]
 ====
-When re-scaffolding, the scaffolder acts on new combinations of channel, server, and `publish` operation groupings in the specification and does not re-scaffold combinations that are not new.
+When re-scaffolding, the scaffolder acts on new combinations of channel, server, and *publish* operation groupings in the specification and does not re-scaffold combinations that are not new.
```
Comment (simoneegeib) [inline modules/ROOT/pages/imp-asyncapi.adoc]: Not only publish. Even though we're not adding a new flow for subscribe operations, we still update the APIkit module in teh project

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 1 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc]: It was taken from the enhanced doc, but according to Aby, it’s untrue. Removed.

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 2 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc:46]: OK! Thanks!

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 3 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc]: OK! Thanks!

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 4]
```diff
@@ -30,20 +30,21 @@ When scaffolding an AsyncAPI specification into a Mule project (implementation o
...
 * Produces the configuration properties file `dev-properties.properties` file in the `src/main/resources` directory of your Mule project
 +
 See <<ex-config-properties>>.
 * Makes the APIkit for AsyncAPI Module (AsyncAPI Module) available for use in the project 
 +
 The module provides the following operations:
 +
-** *Publish* (`<apikit-asyncapi:publish/>`): AsyncAPI `subscribe` operations in the specification are available for configuration as AsyncAPI *Publish* operations. See the <<ex-flows-implemented, implemented example>> in Example: Scaffolded Flows.
-** *Message Listener* (`<apikit-asyncapi:message-listener/>`): AsyncAPI `publish` operations in the specification are configurable *Message Listener* (_not_ *Publish*) operations. See examples in <<ex-flows-scaffolded>>.
+** *Subscribe* (`<apikit-asyncapi:publish/>`): AsyncAPI *subscribe* operations in the specification are available for configuration as AsyncAPI *Subscribe* operations. See the <<ex-flows-implemented, implemented example>> in Example: Scaffolded Flows.
```
Comment (aby-philip-sf) [inline modules/ROOT/pages/imp-asyncapi.adoc]: AsyncAPI *subscribe* operations in the specification are available for configuration as AsyncAPI *Publish* operations.

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 5]
```diff
@@ -30,20 +30,21 @@ When scaffolding an AsyncAPI specification into a Mule project (implementation o
...
+** *Message Listener* (`<apikit-asyncapi:message-listener/>`): AsyncAPI *publish* operations in the specification are configurable *Message Listener* operations. The system publishes messages to a channel, and the Mule application listens for these messages. See examples in <<ex-flows-scaffolded>>.
```
Comment (aby-philip-sf) [inline modules/ROOT/pages/imp-asyncapi.adoc]: `are configurable` to be changed to `are configurable as`

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 6]
```diff
@@ -30,20 +30,21 @@ When scaffolding an AsyncAPI specification into a Mule project (implementation o
...
 +
-The scaffolder transforms each `publish` operation into a *Message Listener* operation.
+The scaffolder transforms each *publish* and *subscribe* operation into a *Message Listener* operation allowing the Mule application to read from the specified channel and process the published messages.
```
Comment (aby-philip-sf) [inline modules/ROOT/pages/imp-asyncapi.adoc]: omit the `and *subscribe*`
Only the publish operations are trasformed as Message Listener in the mule application

**Review by aby-philip-sf — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 4 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc]: OK! Thanks!

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 5 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc]: OK! Thanks!

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 6 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc]: OK! Thanks!

**Review by aby-philip-sf — DISMISSED** [review]
LGTM

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 7]
```diff
@@ -30,20 +30,21 @@ When scaffolding an AsyncAPI specification into a Mule project (implementation o
 * Introspects the AsyncAPI specification 
 * Considers one specification at a time
 * Treats the imported specification as a modification of an existing specification and evaluates it for rescaffolding
-* Creates a new Mule project with a separate flow for each `publish` operation in the specification
+* Creates a new Mule project with a separate flow for each *publish* operation in the specification
```
Comment (sathya0) [inline modules/ROOT/pages/imp-asyncapi.adoc:33]:
```suggestion
* Creates a new Mule project with a separate flow for each *Publish* operation in the specification
```

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 8]
```diff
@@ -30,20 +30,21 @@ When scaffolding an AsyncAPI specification into a Mule project (implementation o
...
 +
 See <<ex-flows-scaffolded>>.
+* Generates an empty XML file with the name of your Mule project if the AsyncAPI spec doesn’t include a *publish* operation
```
Comment (sathya0) [inline modules/ROOT/pages/imp-asyncapi.adoc:36]:
```suggestion
* Generates an empty XML file with the name of your Mule project if the AsyncAPI spec doesn’t include a *Publish* operation
```

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 9]
```diff
@@ -30,20 +30,21 @@ When scaffolding an AsyncAPI specification into a Mule project (implementation o
...
 * Makes the APIkit for AsyncAPI Module (AsyncAPI Module) available for use in the project 
 +
 The module provides the following operations:
 +
-** *Publish* (`<apikit-asyncapi:publish/>`): AsyncAPI `subscribe` operations in the specification are available for configuration as AsyncAPI *Publish* operations. See the <<ex-flows-implemented, implemented example>> in Example: Scaffolded Flows.
-** *Message Listener* (`<apikit-asyncapi:message-listener/>`): AsyncAPI `publish` operations in the specification are configurable *Message Listener* (_not_ *Publish*) operations. See examples in <<ex-flows-scaffolded>>.
+** *Subscribe* (`<apikit-asyncapi:publish/>`): AsyncAPI *subscribe* operations in the specification are available for configuration as AsyncAPI *Publish* operations. See the <<ex-flows-implemented, implemented example>> in Example: Scaffolded Flows.
+** *Message Listener* (`<apikit-asyncapi:message-listener/>`): AsyncAPI *publish* operations in the specification are configurable as *Message Listener* operations. The system publishes messages to a channel, and the Mule application listens for these messages. See examples in <<ex-flows-scaffolded>>.
```
Comment (sathya0) [inline modules/ROOT/pages/imp-asyncapi.adoc:45]:
```suggestion
** *Message Listener* (`<apikit-asyncapi:message-listener/>`): AsyncAPI *Publish* operations in the specification are configurable as *Message Listener* operations. The system publishes messages to a channel, and the Mule application listens for these messages. See examples in <<ex-flows-scaffolded>>.
```

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 10]
```diff
@@ -30,20 +30,21 @@ When scaffolding an AsyncAPI specification into a Mule project (implementation o
...
 +
-The scaffolder transforms each `publish` operation into a *Message Listener* operation.
+The scaffolder transforms each *publish* operation into a *Message Listener* operation allowing the Mule application to read from the specified channel and process the published messages.
```
Comment (sathya0) [inline modules/ROOT/pages/imp-asyncapi.adoc:47]:
```suggestion
The scaffolder transforms each *Publish* operation into a *Message Listener* operation allowing the Mule application to read from the specified channel and process the published messages.
```

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 11]
```diff
@@ -72,7 +73,7 @@ You can scaffold an AsyncAPI specification into a new or existing Mule project:
 [[important-rescaffold]]
 [IMPORTANT]
 ====
-When re-scaffolding, the scaffolder acts on new combinations of channel, server, and `publish` operation groupings in the specification and does not re-scaffold combinations that are not new.
+When re-scaffolding, the scaffolder acts on new combinations of channel, server, and *publish* and *subscribe* operations groupings in the specification and doesn’t re-scaffold combinations that aren’t new.
```
Comment (sathya0) [inline modules/ROOT/pages/imp-asyncapi.adoc:76]:
```suggestion
When re-scaffolding, the scaffolder acts on new combinations of channel, server, and *Publish* and *Subscribe* operations groupings in the specification and doesn’t re-scaffold combinations that aren’t new.
```

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 12]
```diff
@@ -181,7 +182,7 @@ The examples describe an API specification and implementation for processing ord
 The AsyncAPI specification defines an order processing API that uses Anypoint MQ and Kafka protocols.
 * <<ex-flows-scaffolded>>
 +
-The scaffolder generates a `flow.xml` file with a flow for each `publish` operation in the AsyncAPI specification (for re-scaffolding behavior, see <<scaffold-procedures>>). The example describes the generated flows and configures components that provide order-processing logic to some of the flows. 
+The scaffolder generates a `flow.xml` file with a flow for each *publish* operation in the AsyncAPI specification (for re-scaffolding behavior, see <<scaffold-procedures>>). The example describes the generated flows and configures components that provide order-processing logic to some of the flows.
```
Comment (sathya0) [inline modules/ROOT/pages/imp-asyncapi.adoc:185]:
```suggestion
The scaffolder generates a `flow.xml` file with a flow for each *Publish* operation in the AsyncAPI specification (for re-scaffolding behavior, see <<scaffold-procedures>>). The example describes the generated flows and configures components that provide order-processing logic to some of the flows. 
```

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 13]
```diff
@@ -196,7 +197,7 @@ The subflow (`<sub-flow/>`) in `notifications.xml` uses xref:email-connector::in
...
+The following example defines an API specification that uses Anypoint MQ and Kafka protocols. The specification defines https://www.asyncapi.com/docs/concepts/channel[channels] for orders (new, confirmed, and cancelled orders) that use Anypoint MQ, and it defines a channel that uses Kafka for backorders (items that are not in stock). Each channel includes a *publish* and *subscribe* operation. The https://www.asyncapi.com/docs/concepts/server[server] settings in the specification define message brokers that determine the connectors to use (_indirectly_) when publishing events or subscribing to events through operations in the AsyncAPI module.
```
Comment (sathya0) [inline modules/ROOT/pages/imp-asyncapi.adoc:200]:
```suggestion
The following example defines an API specification that uses Anypoint MQ and Kafka protocols. The specification defines https://www.asyncapi.com/docs/concepts/channel[channels] for orders (new, confirmed, and cancelled orders) that use Anypoint MQ, and it defines a channel that uses Kafka for backorders (items that are not in stock). Each channel includes a *Publish* and *Subscribe* operation. The https://www.asyncapi.com/docs/concepts/server[server] settings in the specification define message brokers that determine the connectors to use (_indirectly_) when publishing events or subscribing to events through operations in the AsyncAPI module.
```

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 14]
```diff
@@ -357,7 +358,7 @@ You can discover the URL for an MQ broker from the region that you select when c
 [[ex-flows-scaffolded]]
 === Example: Scaffolded Flows
 
-When scaffolding the specification into an implementation project, Anypoint Code Builder creates a flow (`<flow/>`) for each `publish` operation and transforms the `publish` operation into a *Message Listener* from the AsyncAPI module for APIkit (`<apikit-asyncapi:message-listener/>`). Each flow also includes a Logger. For more information about the module, see xref:apikit::apikit-asyncapi-module-reference.adoc[].
+When scaffolding the specification into an implementation project, Anypoint Code Builder creates a flow (`<flow/>`) for each *publish* operation and transforms the *publish* operation into a *Message Listener* from the AsyncAPI module for APIkit (`<apikit-asyncapi:message-listener/>`). Each flow also includes a Logger. For more information about the module, see xref:apikit::apikit-asyncapi-module-reference.adoc[].
```
Comment (sathya0) [inline modules/ROOT/pages/imp-asyncapi.adoc]:
```suggestion
When scaffolding the specification into an implementation project, Anypoint Code Builder creates a flow (`<flow/>`) for each *Publish* operation and transforms the *Publish* operation into a *Message Listener* source from the AsyncAPI module for APIkit (`<apikit-asyncapi:message-listener/>`). Each flow also includes a *Logger* component. For more information about the module, see xref:apikit::apikit-asyncapi-module-reference.adoc[].
```

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 15]
```diff
@@ -696,8 +697,8 @@ When triggered by the Flow Reference component (`<flow-ref/>`) in the `LISTEN:li
 --
 [calloutlist]
 . Transforms payload output from the `LISTEN:listen-order-confirmations` flow into Mule variables that capture confirmation message data
-. Posts the Slack confirmation message by using the `<slack:post-message/>` operation from the Slack connector 
-. Emails the confirmation message by using the `<email:send/>` operation from the Email connector
+. Posts the Slack confirmation message by using the *<slack:post-message/>*`* operation from the Slack connector
```
Comment (sathya0) [inline modules/ROOT/pages/imp-asyncapi.adoc]:
```suggestion
. Posts the Slack confirmation message by using the *<slack:post-message/>*`* operation from Slack Connector 
```

File: modules/ROOT/pages/imp-asyncapi.adoc [hunk 16]
```diff
@@ -696,8 +697,8 @@ When triggered by the Flow Reference component (`<flow-ref/>`) in the `LISTEN:li
 --
 [calloutlist]
 . Transforms payload output from the `LISTEN:listen-order-confirmations` flow into Mule variables that capture confirmation message data
-. Posts the Slack confirmation message by using the `<slack:post-message/>` operation from the Slack connector 
-. Emails the confirmation message by using the `<email:send/>` operation from the Email connector
+. Posts the Slack confirmation message by using the *<slack:post-message/>*`* operation from the Slack connector 
+. Emails the confirmation message by using the *<email:send/>* operation from the Email connector
```
Comment (sathya0) [inline modules/ROOT/pages/imp-asyncapi.adoc]:
```suggestion
. Emails the confirmation message by using the *<email:send/>* operation from Email Connector
```

**Review by sathya0 — DISMISSED** [review]

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 7 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc:33]: In the AsyncAPI docs, `publish` and `subscribe` operations are not capitalized. Thanks!

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 8 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc:36]: In the AsyncAPI docs, `publish` and `subscribe` operations are not capitalized. Thanks!

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 9 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc:45]: In the AsyncAPI docs, `publish` and `subscribe` operations are not capitalized. Thanks!

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 10 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc:47]: In the AsyncAPI docs, `publish` and `subscribe` operations are not capitalized. Thanks!

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 11 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc:76]: In the AsyncAPI docs, `publish` and `subscribe` operations are not capitalized. Thanks!

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 12 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc:185]: In the AsyncAPI docs, `publish` and `subscribe` operations are not capitalized. Thanks!

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 13 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc:200]: In the AsyncAPI docs, `publish` and `subscribe` operations are not capitalized. Thanks!

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 14 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc]: Great catch! Thanks!

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 15 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc]: OK! Thanks!

File: modules/ROOT/pages/imp-asyncapi.adoc (hunk 16 above)
Comment (Crispy-Salesforce) [reply modules/ROOT/pages/imp-asyncapi.adoc]: OK! Thanks!

**Review by sathya0 — APPROVED** [review]

**Pull request merged**